repository/src/main/java/org/apache/jackrabbit/core/security/authorization/acl/PentahoACLProvider.java

```diff
@@ -121,6 +120,39 @@ protected void updateRootAcl( SessionImpl systemSession, ACLEditor editor ) thro
   }
 
   /**
+   * Returns true if the root acl needs updating (if the JCR_READ_ACCESS_CONTROL privilege is missing from the
+   * 'everyone' principle) and false otherwise.
+   */
+  protected boolean requireRootAclUpdate( ACLEditor editor ) throws RepositoryException {
+    final String rootPath = session.getRootNode().getPath();
+    final AccessControlPolicy[] acls = editor.getPolicies( rootPath );
+    if ( acls != null &&  acls.length > 0 ) {
+      final PrincipalManager pMgr = session.getPrincipalManager();
+      final AccessControlManager acMgr = session.getAccessControlManager();
+      final Privilege jcrReadAccessControlPriv =  acMgr.privilegeFromName( Privilege.JCR_READ_ACCESS_CONTROL );
+      final Principal everyone = pMgr.getEveryone();
+      final AccessControlList acList = (AccessControlList) acls[0];
+      final AccessControlEntry[] acEntries = acList.getAccessControlEntries();
+      if ( acEntries != null ) {
+        for ( AccessControlEntry acEntry : acEntries ) {
+          if ( acEntry.getPrincipal() != null && acEntry.getPrincipal().equals( everyone ) ) {
+            if ( acEntry.getPrivileges() != null ) {
+              for ( Privilege privilege : acEntry.getPrivileges() ) {
+                if ( jcrReadAccessControlPriv.equals( privilege ) ) {
+                  // If the everyone principal already has the JCR_READ_ACCESS_CONTROL privilege, there's no need to
+                  // update ACL
+                  return false;
+                }
+              }
+            }
+          }
```

Comment:
Not sure if the logic is the same as below?

if ( acEntry.getPrincipal() != null && acEntry.getPrincipal().equals( everyone ) && acEntry.getPrivileges() != null  ) {
  return Arrays.asList( acEntry.getPrivileges() ).contains( jcrReadAccessControlPriv );
}

Reply:
Sure - that works too